tutorials/sphinx-tutorials/multiagent_ppo.py

```diff
@@ -0,0 +1,789 @@
+    n_agent_outputs=1,  # 1 value per agent
+    n_agents=env.n_agents,
+    centralised=mappo,
+    share_params=share_parameters_critic,
+    device=device,
+    depth=2,
+    num_cells=256,
+    activation_class=torch.nn.Tanh,
+)
+
+critic = TensorDictModule(
+    module=critic_net,
+    in_keys=[("agents", "observation")],
+    out_keys=[("agents", "state_value")],
+)
+
+######################################################################
+# let's try our policy and critic modules. As we said earlier, the usage of
+# :class:`TensorDictModule` makes it possible to directly read the output
+# of the environment to run these modules, as they know what information to read
+# and where to write it:
+#
+# **From this point on, the multi-agent-specific components have been instantiated, and we will simply use the same
+# components as in single-agent learning. Isn't this fantastic?**
+#
+print("Running policy:", policy(env.reset()))
+print("Running value:", critic(env.reset()))
+
+######################################################################
+# Data collector
+# --------------
+#
+# TorchRL provides a set of :class:`DataCollector` classes. Briefly, these
+# classes execute three operations: reset an environment, compute an action
+# using the policy and the latest observation, execute a step in the environment, and repeat
+# the last two steps until the environment signals a stop (or reaches a done
+# state).
+#
+# We will use the simplest possible data collector, which has the same output as an environment rollout,
+# with the only difference that it will auto reset done states until the desired frames are collected.
+#
+collector = SyncDataCollector(
+    env,
+    policy,
+    device=vmas_device,
+    storing_device=device,
+    frames_per_batch=frames_per_batch,
+    total_frames=total_frames,
+)
+
+######################################################################
+# Replay buffer
+# -------------
+#
+# Replay buffers are a common building piece of off-policy RL algorithms.
+# In on-policy contexts, a replay buffer is refilled every time a batch of
+# data is collected, and its data is repeatedly consumed for a certain number
+# of epochs.
+#
+# Using a replay buffer for PPO is not mandatory and we could simply
+# use the collected data online, but using these classes
+# makes it easy for us to build the inner training loop in a reproducible way.
+#
+
+replay_buffer = ReplayBuffer(
+    storage=LazyTensorStorage(
+        frames_per_batch, device=device
+    ),  # We store the frames_per_batch collected at each iteration
+    sampler=SamplerWithoutReplacement(),
+    batch_size=minibatch_size,  # We will sample minibatches of this size
+)
+
+######################################################################
+# Loss function
+# -------------
+#
+# The PPO loss can be directly imported from torchrl for convenience using the
+# :class:`ClipPPOLoss` class. This is the easiest way of utilizing PPO:
```

Comment:
```suggestion
# :class:`~.objectives.ClipPPOLoss` class. This is the easiest way of utilizing PPO:
```

Reply:
We should decide if we want british or US spelling: utilise or utilize
We used flavour before, so UK